### ISSUE TITLE
Certificate for <developer.huawei.com> doesn't match any of the subject alternative names

### ISSUE THREAD

**Issue by kevinmcody** [transistorsoft/cordova-background-geolocation-lt#1439]
## Your Environment
* Plugin version: 4.16.1
* Platform: Android
* OS version: Windows 11
* Device manufacturer / model: N/A
* Cordova version (`cordova -v`): 12.0.0
* Cordova platform version (`cordova platform ls`): 12.0.1
* Plugin config provided to `#ready`: N/A

## Expected Behavior
I should be able to build the project

## Actual Behavior
The project fails to build, with many errors regarding developer.huawei.com, all similar to the what I pasted in the "logs" section below:

## Steps to Reproduce
1. From a fresh cordova project
2. cordova platform add android
3. cordova plugin add cordova-background-geolocation-lt
4. cordova build android --debug

## Context
Build the project.  Note that this worked perhaps a month or two ago.

## Debug logs

<details>
	<summary>Logs</summary>

```<!-- Syntax highlighting:  DO NOT REMOVE -->
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Could not resolve com.huawei.hms:availableupdate:6.5.0.300.
     Required by:
         project :app > com.huawei.hms:location:6.9.0.300 > com.huawei.hms:base:6.5.0.300
      > Could not resolve com.huawei.hms:availableupdate:6.5.0.300.
         > Could not get resource 'https://developer.huawei.com/repo/com/huawei/hms/availableupdate/6.5.0.300/availableupdate-6.5.0.300.pom'.
            > Could not GET 'https://developer.huawei.com/repo/com/huawei/hms/availableupdate/6.5.0.300/availableupdate-6.5.0.300.pom'.
               > Certificate for <developer.huawei.com> doesn't match any of the subject alternative names: [a248.e.akamai.net, *.akamaized.net, *.akamaized-staging.net, *.akamaihd.net, *.akamaihd-staging.net]
```

</details>

**Comment by kevinmcody** [comment]
Thanks.  

I worked around it temporarily by doing something my IT security folks should never know about (a Gradle plugin that ignores SSL errors while pulling in external repos):  https://stackoverflow.com/questions/51658086/how-to-configure-gradle-to-bypass-ssl-certificate-validation

(As for Cordova, LOL I 100% agree!  Sadly the suits that manage my time have other priorities)

**Comment by kevinmcody** [comment]
Update - the temporary workaround did not, in fact, work. Nor did several other attempts I made to force Gradle / Java to ignore cert validation errors. 

Any luck reproducing this?

**Comment by christocracy** [comment]
I just generate a fresh Cordova app, added the plugin, pasted the example and it works fine.  The plugin requests permissions at app launch (you're probably not even getting the android app to build)

```
$ cordova platforms
Installed platforms:
  android 12.0.1
```

**Comment by kevinmcody** [comment]
I'm grasping at very thin straws right now:  what version of Gradle are you using?

**Comment by christocracy** [comment]
platforms/android/gradle/gradle-wrapper.properties

```
distributionUrl=https://services.gradle.org/distributions/gradle-7.6-all.zip
```

**Comment by kevinmcody** [comment]
I think I fixed this by changing JDKs, which sounds kinda crazy but hear me out:

The Huawei stuff that was being pulled in was all at developer.huawei.com, which apparently redirects to a CDN called "akamai.net".  When I went to the repos directly in a browser I was able to download them without issue; it was just Gradle running in Powershell that was complaining.  I'm thinking maybe the Java HttpClient simply was not smart enough to navigate the redirects.

I switched my JAVA_HOME environment variable from OpenJDK11 to the JetBrains Java Runtime that ships with Android Studio,  and now it successfully pulls in all the repos and builds.
